Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/tasks/service/PersistedTasksServiceTest.kt

```diff
@@ -852,36 +909,109 @@ class PersistedTasksServiceTest {
         val tasks = taskListWithSomeCompleted()
         putTaskListIntoRemoteDataSource(tasks)
         taskLocalDataSubject.onCompleted()
-        service.tasksEvent.subscribe(TestObserver<Event<Tasks>>())
         `when`(remoteDataSource.clearCompletedTasks()).thenReturn(Observable.just(remoteTasksWithCompletedTasksRemoved()))
 
         val testObserver = subscribeToTasksEvent()
 
-        service.clearCompletedTasks().call()
+        service.clearCompletedTasks()
 
         testObserver.assertReceivedOnNext(listOf(
                 idleEventWith(asSyncedTasks(tasks)), // ideally, we would ignore this
-                loadingEventWith(tasksWithCompletedMarkedDeleted()),
+                loadingEventWith(asSyncedTasks(tasks)),
+                loadingEventWith(tasksWithCompletedMarkedDeleted(TEST_TIME)),
                 loadingEventWith(asSyncedTasks(remoteTasksWithCompletedTasksRemoved())),
                 idleEventWith(asSyncedTasks(remoteTasksWithCompletedTasksRemoved()))
         ))
     }
 
     @Test
+    fun given_WeHavePendingTaskInTheService_on_ClearCompletedTasks_it_ShouldNotUpdatePendingTask() {
```

Comment:
Is there any way we can make these couple of tests a little bit more readable, maybe just with a given?